syntax = "proto3";

enum Genre {
  UNKNOWN_GENRE = 0;
  FICTION = 1;
  NON_FICTION = 2;
  SCIENCE_FICTION = 3;
}

enum Status {
  UNKNOWN_STATUS = 0;
  AVAILABLE = 1;
  TAKEN = 2;
}

message Book {
  string isbn = 1;
  string title = 2;
  string author = 3;
  Genre genre = 4;
  int32 publishing_year = 5;
}

message InventoryItem {
  string inventory_number = 1;
  oneof item_type {
    Book book = 2;
  }
  Status status = 3;
}

message StringMessage {
  string value = 1;
}


service InventoryService {
  rpc CreateBook(Book) returns (StringMessage) {}
  rpc GetBook(StringMessage) returns (Book) {}
}